HeterogeneousCore/ROCmServices/bin/BuildFile.xml

```diff
@@ -0,0 +1,9 @@
+<iftool name="rocm">
+  <bin name="rocmComputeCapabilities" file="rocmComputeCapabilities.cpp isRocmDeviceSupported.hip.cc">
+    <use name="rocm"/>
+  </bin>
+
+  <bin name="rocmIsEnabled" file="rocmIsEnabled.cpp isRocmDeviceSupported.hip.cc">
```

Comment:
I had long thought we would (eventually) always build `cudaIsEnabled`, `rocmIsEnabled` etc, and in case the corresponding platform is not available on a given `SCRAM_ARCH`, the program would just return `EXIT_FAILURE`.

On the other hand, for the use in python configuration, the failure to run such a program from `$PATH` would be automatically interpreted as "platform is unavailable", so maybe this current approach of not building the `*IsEnabled` program when the platform is not available is ok.

Thoughts?

Reply:
I'm fine with always building something, that would just return "false".
Does SCRAM support something like `<ifnottool>` ?

Reply:
> Does SCRAM support something like <ifnottool> ?

yes SCRAM does support it but you need to use `!` e.g `<iftool name="!rocm"/>`

Reply:
mhm, I tried this, but I couldn't get it to work :-(
do we have any examples ?

Reply:
ah my bad, `!` was supported for all the other filters (e.g ifrelease, ifarch etc) but not for iftool. https://github.com/cms-sw/cmsdist/pull/8272 should fix this

Reply:
thanks !

Reply:
@smuzaffar ok, so with this can have two different source files, one that uses ROCm and one that does not:
```xml
<iftool name="rocm">
  <bin name="rocmIsEnabled" file="rocmIsEnabled.cpp isRocmDeviceSupported.hip.cc">
    <use name="rocm"/>
  </bin>
</iftool>
<iftool name="!rocm">
  <bin name="rocmIsEnabled" file="rocmIsEnabled_fallback.cpp">
  </bin>
</iftool>
```
Is that OK ?

Reply:
you can use `<else/>` tag e.g. the following should work with current SCRAM

```
<iftool name="rocm">
  <bin name="rocmIsEnabled" file="rocmIsEnabled.cpp isRocmDeviceSupported.hip.cc">
    <use name="rocm"/>
  </bin>
<else/>
  <bin name="rocmIsEnabled" file="rocmIsEnabled_fallback.cpp">
  </bin>
</iftool>
```